pkg/network/encoding/protobuf.go

```diff
@@ -8,24 +8,19 @@ package encoding
 import (
 	model "github.com/DataDog/agent-payload/v5/process"
 	"github.com/gogo/protobuf/proto"
-
-	"github.com/DataDog/datadog-agent/pkg/network"
 )
 
 // ContentTypeProtobuf holds the HTML content-type of a Protobuf payload
 const ContentTypeProtobuf = "application/protobuf"
 
 type protoSerializer struct{}
 
-func (protoSerializer) Marshal(conns *network.Connections) ([]byte, error) {
-	payload := modelConnections(conns)
-	buf, err := proto.Marshal(payload)
-	returnToPool(payload)
-	return buf, err
+func (protoSerializer) Marshal(conns *model.Connections) ([]byte, error) {
+	return proto.Marshal(conns)
 }
 
 func (protoSerializer) Unmarshal(blob []byte) (*model.Connections, error) {
-	conns := new(model.Connections)
+	conns := connsPool.Get().(*model.Connections)
```

Comment:
I think this change (use of pools) should be done in a different PR as it is not related to batching

Reply:
You are right, I will handle it